src/main/java/org/nibor/autolink/LinkExtractor.java

```diff
@@ -27,11 +27,13 @@ public static Builder builder() {
 
     /**
      * Extract the links from the input text. Can be called multiple times with different inputs (thread-safe).
-     *
-     * @param input the input text, must not be {@code null}
+     * @param input - the input text
      * @return a lazy iterable for the links in order that they appear in the input, never {@code null}
+     * @throws NullPointerException if the input argument is null 
      */
     public Iterable<LinkSpan> extractLinks(final CharSequence input) {
+    	if (input == null)
+    		throw new NullPointerException("input argument can't be null.");
```

Comment:
Good idea! Could you change this to use [Objects.requireNonNull](https://docs.oracle.com/javase/7/docs/api/java/util/Objects.html#requireNonNull%28T,%20java.lang.String%29) instead?